src/sparc/client/services/metadata.py

```diff
@@ -0,0 +1,277 @@
+import json
+import logging
+from configparser import SectionProxy
+from typing import List, Optional, Union
+
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+from ._default import ServiceBase
+
+
+class MetadataService(ServiceBase):
+    """A wrapper for the Elasticsearch Metadata library
+
+    Parameters:
+    -----------
+    config : dict
+        A configuration containing necessary API key (scicrunch_api_key).
+    connect : bool
+        Not needed with REST metadata services.
+
+    Attributes:
+    -----------
+    default_headers : dict
+        A dictionary with headers to make HTTP requests.
+    host_api : str
+        A default HTTP address of the SciCrunch Elasticsearch API endpoint.
+
+    Methods:
+    --------
+    get_profile() -> str
+        Returns the currently used API Key.
+    set_profile() -> str
+        Changes the API Key.
+    close() : None
+        Not needed with REST metadata services.
+    getURL(...) : dict
+        Supporting function to retrieve data from REST endpoint via GET
+        This support Elasticsearch URL based queries
+    postURL(...) : dict
+        Supporting function to retrieve data from REST endpoint
+        This supports Elasticsearch JSON queries
+    list_datasets(...) : dict
+        Returns a dictionary with datasets metadata.
+    search_datasets(...) : dict
+        Returns a dictionary with datasets matching search criteria.
+
+    """
+
+    default_headers = {
+        "Content-Type": "application/json",
+        "Accept": "application/json; charset=utf-8",
+    }
+
+    host_api = "https://scicrunch.org/api/1/elastic"
+
+    scicrunch_api_key: str = None
+    profile_name: str = None
+
+    def __init__(
+        self, config: Optional[Union[dict, SectionProxy]] = None, connect: bool = False
+    ) -> None:
+        logging.info("Initializing SPARC K-Core Elasticsearch services...")
+        logging.debug(str(config))
+
+        self.host_api = "https://scicrunch.org/api/1/elastic"
```

Comment:
This has already been declared.

Reply:
Removed straggler host_api = outside of _init_